vignettes/fr/datatable-secondary-indices-and-auto-indexing.Rmd

```diff
@@ -0,0 +1,332 @@
+
+#### -- Il ne peut y avoir qu'une seule `clé` au maximum
+
+Maintenant, si nous voulons répéter la même opération mais sur la colonne `dest` à la place, pour la valeur "LAX", alors nous devons utiliser `setkey()`, *une fois de plus*.
+
+```{r, eval = FALSE}
+## pas exécuté
+setkey(flights, dest)
+flights["LAX"]
+```
+
+Et cela réordonne les `vols` par `dest`, *encore une fois*. Ce que nous aimerions vraiment, c'est pouvoir effectuer le sous-ensemble rapidement en éliminant l'étape de réorganisation.
+
+Et c'est précisément ce que permettent les *indices secondaires* !
+
+#### -- Les indices secondaires peuvent être réutilisés
+
+Comme il peut y avoir plusieurs indices secondaires et que la création d'un indice est aussi simple que le stockage du vecteur d'ordre en tant qu'attribut, cela nous permet même d'éliminer le temps nécessaire pour recalculer le vecteur d'ordre si un indice existe déjà.
+
+#### -- Le nouvel argument `on` permet une syntaxe plus propre ainsi que la création et la réutilisation automatiques d'indices secondaires
+
+Comme nous le verrons dans la section suivante, l'argument `on` présente plusieurs avantages :
+
+#### Argument `on`
+
+* permet d’effectuer des sous-ensembles en calculant les indices secondaires à la volée. Cela évite d'avoir à faire `setindex()` à chaque fois.
+
+* permet de réutiliser facilement les indices existants en vérifiant simplement les attributs.
+
+* permet une syntaxe plus propre en intégrant dans la syntaxe les colonnes sur lesquelles le sous-ensemble est effectué. Le code est ainsi plus facile à suivre lorsqu'on le consulte ultérieurement.
+
+    Notez que l'argument `on` peut également être utilisé pour les sous-ensembles à clés. En fait, nous encourageons à fournir l'argument `on` même lorsque le sous-ensemble utilise des clés pour une meilleure lisibilité.
+
+# 
+
+## 2. Sous-ensemble rapide utilisant l'argument `on` et les indices secondaires
+
+### a) Sous-ensembles rapides dans `i`
+
+#### -- Sous-ensemble de toutes les lignes où l'aéroport d'origine correspond à *"JFK"* en utilisant `on`
+
+```{r}
+flights["JFK", on = "origin"]
+
+## ou alors
+# flights[.("JFK"), on = "origin"] (or)
+# flights[list("JFK"), on = "origin"]
+```
+
+* Cette instruction effectue également une recherche binaire rapide basée sur le sous-ensemble, en calculant l'index à la volée. Cependant, notez qu'elle n'enregistre pas automatiquement l'index en tant qu'attribut. Cela pourrait changer à l'avenir.
+
+* Si nous avions déjà créé un index secondaire en utilisant `setindex()`, alors `on` le réutiliserait au lieu de le (re)calculer. Nous pouvons le voir en utilisant `verbose = TRUE`:
+
+    ```{r}
+    setindex(flights, origin)
+    flights["JFK", on = "origin", verbose = TRUE][1:5]
+    ```
+
+#### -- Comment puis-je faire un sous-ensemble basé sur les colonnes `origin` *et* `dest` ?
+
+Par exemple, si nous voulons un sous-ensemble combinant `"JFK" et "LAX"`, alors :
```

Comment:
`` `"JFK"` et `"LAX"` ``

Reply:
Well, in the original English version we have:

````
#### -- How can I subset based on `origin` *and* `dest` columns?

For example, if we want to subset `"JFK", "LAX"` combination, then:

```{r}
flights[.("JFK", "LAX"), on = c("origin", "dest")][1:5]
```
````

So, that seems correct to me.

Reply:
in the original a comma is used (which could be interpreted as incomplete code, so ok in backticks) but `et` is a French word (not code, so should not be inside backticks)
probably better to change both English and French versions to `c("JFK","LAX")` what do you think?

Reply:
Ah yes ! I got it now. You are right. I'll make the change.